for-research/phoenix/get_json_phoenix.py

```diff
@@ -32,159 +32,243 @@
-                                                'mass':branch_forms['AnalysisPhotonsAuxDyn.m'],
-                                                'Cluster_Links':branch_forms['AnalysisPhotonsAuxDyn.caloClusterLinks']},
-                                               'AnalysisPhotons', 'PtEtaPhiMLorentzVector')
+        possible_branches = {
+            'pt': 'AnalysisPhotonsAuxDyn.pt',
+            'eta': 'AnalysisPhotonsAuxDyn.eta',
+            'phi': 'AnalysisPhotonsAuxDyn.phi',
+            'mass': 'AnalysisPhotonsAuxDyn.m',
+            'Cluster_Links': 'AnalysisPhotonsAuxDyn.caloClusterLinks'
+        }
+        photons = create_collection(branch_forms, possible_branches, 'AnalysisPhotons', 'PtEtaPhiMLorentzVector')
+        if photons:
+            output['AnalysisPhotons'] = photons
+
         # Electrons - four vector plus cluster and track links
-        output['AnalysisElectrons'] = zip_forms( {'pt':branch_forms['AnalysisElectronsAuxDyn.pt'],
-                                                'eta':branch_forms['AnalysisElectronsAuxDyn.eta'],
-                                                'phi':branch_forms['AnalysisElectronsAuxDyn.phi'],
-                                                'mass':branch_forms['AnalysisElectronsAuxDyn.m'],
-                                                'Cluster_Links':branch_forms['AnalysisElectronsAuxDyn.caloClusterLinks'],
-                                                'Track_Links':branch_forms['AnalysisElectronsAuxDyn.trackParticleLinks']},
-                                               'AnalysisElectrons', 'PtEtaPhiMLorentzVector')
+        possible_branches = {
+            'pt': 'AnalysisElectronsAuxDyn.pt',
+            'eta': 'AnalysisElectronsAuxDyn.eta',
+            'phi': 'AnalysisElectronsAuxDyn.phi',
+            'mass': 'AnalysisElectronsAuxDyn.m',
+            'Cluster_Links': 'AnalysisElectronsAuxDyn.caloClusterLinks',
+            'Track_Links': 'AnalysisElectronsAuxDyn.trackParticleLinks'
+        }
+        electrons = create_collection(branch_forms, possible_branches, 'AnalysisElectrons', 'PtEtaPhiMLorentzVector')
+        if electrons:
+            output['AnalysisElectrons'] = electrons
+
         # All done putting together our branches! Now just return the output that we've built
         return output
-        
+
     @property
     def behavior(self):
         behavior = {}
         behavior.update(base.behavior)
         behavior.update(vector.behavior)
         return behavior
 
-
 # Similar setup, but for flat ntuples provided using the outreach and education ntuple making framework
 class Flat_NtupleSchema(BaseSchema):
     def __init__(self, base_form):
         super().__init__(base_form)
         self._form['contents'] = self._build_collections(self._form['contents'])
 
     # We don't need ALL of the branches from the ntuple
-    # We only care about the electron variables
+    # We only care about the electron and lepton variables
```

Comment:
Not sure I get it — electron and lepton?

Reply:
no idea where my brain was at.